shared/src/main/scala/StringHelpers.scala

```diff
@@ -15,6 +16,72 @@ object StringHelpers:
       case a: VNum  => a.toInt.toChar.toString
       case a: VList => VList(a.map(chrord)*)
 
+  // https://codegolf.stackexchange.com/a/151721/78850
+  def compressDictionary(s: String)(using ctx: Context): String =
+    val endLength = 2 + ctx.globals.longDictionary.map(_.length).max
+
+    def character(z: Int, c: Char) =
+      val o =
+        if c.toInt == 10 then 95
+        else if ' ' <= c && c <= '~' then c.toInt - 32
+        else throw new Exception(s"Invalid character $c")
+
+      3 * 96 * z + o
+
+    def dictionary(z: Int, w: String, nonempty: Boolean) =
+      var ts = nonempty
+      var subW = w
+      if w.headOption.exists(_ == ' ') then
```

Comment:
Can we assume that `w` is not the empty string? If so, we can do
```suggestion
      if w.charAt(0) == ' ' then
```

Reply:
I honestly have no clue if we can make that assumption. The python I ported had `w[:1] == " "` as the condition

Reply:
Well if we're compressing the empty string, we can probably return early, right? There's no compressing to do if it's already `""`. Not sure where to put that, though

Reply:
Where's hyper when you need him? :p